### PR TITLE
Adopt to psalm level 1 + Cleanup

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/Definitions/CallableDefinition.php [hunk 1]
```diff
@@ -27,18 +27,18 @@ public function resolve(ContainerInterface $container)
     {
         $callable = $this->prepareCallable($this->method, $container);
 
-        return $container->get(Injector::class)->invoke($callable);
+        /** @var Injector $injector */
+        $injector = $container->get(Injector::class);
+
+        return $injector->invoke($callable);
```
Comment (yiiliveext) [inline src/Definitions/CallableDefinition.php:33]: This is superfluous

Comment (vjik) [reply src/Definitions/CallableDefinition.php]: Fixed.

File: src/Factory.php [hunk 2]
```diff
@@ -69,16 +75,15 @@ public function get($id)
 
     /**
      * @param mixed $id
+     *
+     * @throws InvalidConfigException
      */
     public function getDefinition($id): DefinitionInterface
     {
-        if ($this->has($id)) {
```
Comment (samdark) [inline src/Factory.php:75]: Previously it was applied to any existing definition. New behavior is slightly different. Are you sure about it?

Comment (vjik) [reply src/Factory.php:75]: Yes. Definitions setted only with string keys (see method `set(string $id, ...)`).

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/Definitions/CallableDefinition.php [hunk 3]
```diff
@@ -27,18 +27,16 @@ public function resolve(ContainerInterface $container)
...
+        /** @psalm-suppress MixedMethodCall */
         return $container->get(Injector::class)->invoke($callable);
     }
 
-    /**
-     * @param array|callable $callable
-     * @param ContainerInterface $container
-     */
-    private function prepareCallable($callable, ContainerInterface $container): callable
+    private function prepareCallable(callable $callable, ContainerInterface $container): callable
```
[the remark this answers is not in the document]
Comment (vjik) [reply src/Definitions/CallableDefinition.php]: Add test and fix.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**